### PR TITLE
Ch 326

### PR DESCRIPTION
**Pull request by daniel-hlavaty**

### CODE REVIEW

File: src/components/connectButton/ConnectButton.tsx [hunk 1]
```diff
@@ -38,64 +46,129 @@ const ConnectButton = () => {
...
+					await (window.ethereum as any).request({
+						method: 'wallet_switchEthereumChain',
+						params: [{ chainId: network.networkId }]
+					})
+					switchNetwork?.(network.networkId)
+					setIsModalVisible(false)
+				} catch (switchError: any) {
+					if (switchError.code === 4902) {
+						try {
+							await (window.ethereum as any).request({
```
Comment (michalbella) [inline src/components/connectButton/ConnectButton.tsx:62]: You need to provide ethereum type into global.d.ts and inject to window like:
```
interface Window {
	ethereum :  Add type if exist or any
}
```

File: src/components/connectButton/ConnectButton.tsx [hunk 2]
```diff
@@ -38,64 +46,129 @@ const ConnectButton = () => {
...
+				{NETWORK_SWITCHER_SUPPORTED_NETWORKS.map((network) => (
+					<SC.ChainOptions>
+						<SC.ChainRow onClick={() => handleSwitchNetwork(network)} justify={'space-between'}>
+							<Col span={12}>
+								<SC.ChainName style={{ marginBottom: '0px' }}>{network.shortChainName}</SC.ChainName>
+							</Col>
+							<Col flex={'end'}>
+								<SCG.FlexRow>
+									{chain?.id === network.chainId && <SC.Connected>{t('connected')}</SC.Connected>}
+									<SC.Logo alt={''} src={network.icon} />
```
Comment (michalbella) [inline src/components/connectButton/ConnectButton.tsx:107]: add alt text

**Review by michalbella — CHANGES_REQUESTED** [review]

File: src/components/connectButton/ConnectButton.tsx (hunk 1 above)
Comment (daniel-hlavaty) [reply src/components/connectButton/ConnectButton.tsx:61]: fixed

File: src/components/connectButton/ConnectButton.tsx [hunk 3]
```diff
@@ -38,64 +46,129 @@ const ConnectButton = () => {
...
+			if (hasEthereumInjected()) {
+				try {
+					await (window.ethereum as any).request({
+						method: 'wallet_switchEthereumChain',
+						params: [{ chainId: network.networkId }]
+					})
+					switchNetwork?.(network.networkId)
+					setIsModalVisible(false)
+				} catch (switchError: any) {
+					if (switchError.code === 4902) {
```
[the remark this answers is not in the document]
Comment (daniel-hlavaty) [reply src/components/connectButton/ConnectButton.tsx:59]: fixed

**Pull request merged**